chat-client/src/client/mynahUi.ts

```diff
@@ -461,7 +461,24 @@ export const createMynahUi = (
 
         if (isPartialResult) {
             // type for MynahUI differs from ChatResult types so we ignore it
-            mynahUi.updateLastChatAnswer(tabId, { ...chatResultWithoutType, header: header, buttons: buttons })
+            mynahUi.updateLastChatAnswer(tabId, {
+                ...chatResultWithoutType,
+                header: header,
+                buttons: buttons,
+                // TODO: this won't be needed once we have multiple message support
+                ...(type === 'tool'
+                    ? {
+                          header: {
+                              ...header,
+                              // TODO: this is specific to fsWrite tool, modify for other tools as needed
+                              fileList: { ...header?.fileList, fileTreeTitle: '', hideFileCount: true },
```

Comment:
is only this line going to be modified for other tools or all of it? If so, just wondering if it makes sense to pull this out to a different file so that its easy to understand which tools have which interface

Reply:
+1 to separating this logic out. Any tool specific logic should be isolated from the core logic.